pkg/clients/ecr/repository.go

```diff
@@ -125,3 +126,54 @@ func GenerateCreateRepositoryInput(name string, params *v1alpha1.RepositoryParam
 	}
 	return c
 }
+
+// CompareTags compares arrays of v1alpha1.Tag and ecr.Tag
+func CompareTags(tags []v1alpha1.Tag, ecrTags []ecr.Tag) bool {
+	if len(tags) != len(ecrTags) {
+		return false
+	}
+
+	SortTags(tags, ecrTags)
+
+	for i, t := range tags {
+		if t.Key != aws.StringValue(ecrTags[i].Key) || t.Value != aws.StringValue(ecrTags[i].Value) {
+			return false
+		}
+	}
+
+	return true
+}
+
+// SortTags sorts array of v1alpha1.Tag and ecr.Tag on 'Key'
+func SortTags(tags []v1alpha1.Tag, ecrTags []ecr.Tag) {
+	sort.Slice(tags, func(i, j int) bool {
+		return tags[i].Key < tags[j].Key
+	})
+
+	sort.Slice(ecrTags, func(i, j int) bool {
+		return *ecrTags[i].Key < *ecrTags[j].Key
+	})
+}
+
+// DiffTags returns tags that should be added or removed.
+func DiffTags(spec []v1alpha1.Tag, current []ecr.Tag) (addTags []ecr.Tag, remove []string) {
+	addMap := map[string]string{}
```

Comment:
nitpick: we know `addMap` max size
```suggestion
	addMap := make(map[string]string, len(spec))
```